CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_SquadPickupPoint.uc

```diff
@@ -30,49 +29,28 @@ function bool RequiresSquad()
 function SetupPickupLocation(XComGameState NewGameState, XComGameState_CovertAction CovertAction, StrategyCost Cost)
 {
 	local XComGameState_HeadquartersXCom XComHQ;
-	ActionRef = CovertAction.GetReference();
-	// we're intentionally checking if this Region has a XCGS and assigning none if not
-	Region = XComGameState_WorldRegion(`XCOMHISTORY.GetGameStateForObjectID(ActionRef.ObjectID)).GetReference();
+	
+	XComHQ = `XCOMHQ;
 
 	ExfiltrateCost = Cost;
-
 	Location.x = CovertAction.Location.x;
 	Location.y = CovertAction.Location.y;
-	
-	XComHQ = `XCOMHQ;
+	ActionRef = CovertAction.GetReference();
 
+	Region = XComGameState_WorldRegion(`XCOMHISTORY.GetGameStateForObjectID(CovertAction.Region.ObjectID)).GetReference();
 	XComHQ = XComGameState_HeadquartersXCom(NewGameState.ModifyStateObject(class'XComGameState_HeadquartersXCom', XComHQ.GetReference().ObjectID));
 	XComHQ.CrossContinentMission = GetReference();
 }
 
 function FlyToPickupPoint()
 {
-	// if we have a region fly there
-	if (Region.ObjectID != 0)
-	{
-		XComGameState_WorldRegion(`XCOMHISTORY.GetGameStateForObjectID(Region.ObjectID)).ConfirmSelection();
-	}
-	// if we don't just fly to the mission site
-	else
-	{
-		ConfirmSelection();
-	}
+	XComGameState_WorldRegion(`XCOMHISTORY.GetGameStateForObjectID(Region.ObjectID)).ConfirmSelection();
```

Comment:
Set `XComHQ.CrossContinentMission` here (by creating and submitting a new gamestate)